hr_expense_petty_cash/models/account_invoice.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    is_petty_cash = fields.Boolean(
+        string='Petty Cash',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.constrains('invoice_line_ids')
+    def _check_petty_cash_amount(self):
+        petty_cash_env = self.env['petty.cash']
+        for rec in self:
+            if not rec.is_petty_cash:
```

Comment:
Same here, instead of doing this, you could say `for rec in self.filtered()` in order to filter out the records you do not want.